### PR TITLE
Refactor ElasticJobServiceLoader, remove unused code and use Optional…

### LINKED ISSUE #1566: Refactor ElasticJobServiceLoader, remove unused code and use Optional instead of throw JobConfigurationException
Remove unused code in `ElasticJobServiceLoader` and Refactor it method to return a empty Optional instead of throw JobConfigurationException when cannot find the service instance.

### PR DESCRIPTION
**Pull request by wwj-go**
… instead of throw JobConfigurationException (#1566)

Fixes #1566. 
Refactor `org.apache.shardingsphere.elasticjob.infra.spi.ElasticJobServiceLoader`
Changes proposed in this pull request:
- Remove unused code.
- Refactor it method to return an empty Optional instance instead of throw JobConfigurationException when cannot find the service instance.

### CODE REVIEW

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/handler/threadpool/JobExecutorServiceHandlerFactory.java [hunk 1]
```diff
@@ -42,8 +43,11 @@
      */
     public static JobExecutorServiceHandler getHandler(final String type) {
         if (Strings.isNullOrEmpty(type)) {
-            return ElasticJobServiceLoader.getCachedInstance(JobExecutorServiceHandler.class, DEFAULT_HANDLER);
+            return ElasticJobServiceLoader.getCachedTypedServiceInstance(JobExecutorServiceHandler.class, DEFAULT_HANDLER)
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/handler/threadpool/JobExecutorServiceHandlerFactory.java]: Is it necessary to check if default handler present? Why could this happen?

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/spi/ElasticJobServiceLoader.java [hunk 2]
```diff
@@ -106,33 +61,42 @@ private static Object newServiceInstance(final Class<?> clazz) {
...
-     * @param typedService service type
+     * @param typedServiceInterface typed service interface
      * @param type         specific service type
      * @param <T>          specific type of service
      * @return cached service instance
      */
-    public static <T extends TypedSPI> T getCachedInstance(final Class<T> typedService, final String type) {
-        T instance = TYPED_SERVICES.containsKey(typedService) ? (T) TYPED_SERVICES.get(typedService).get(type) : null;
+    public static <T extends TypedSPI> Optional<T> getCachedTypedServiceInstance(final Class<T> typedServiceInterface, final String type) {
+        T instance = TYPED_SERVICES.containsKey(typedServiceInterface) ? (T) TYPED_SERVICES.get(typedServiceInterface).get(type) : null;
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/spi/ElasticJobServiceLoader.java]: Consider making good use of `Optional#map` if you are going to use `Optional`.

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/spi/ElasticJobServiceLoader.java [hunk 3]
```diff
@@ -106,33 +61,42 @@ private static Object newServiceInstance(final Class<?> clazz) {
...
-     * @param typedService service type
+     * @param typedServiceInterface typed service interface
      * @param type         specific service type
      * @param <T>          specific type of service
      * @return specific typed service instance
      */
-    public static <T extends TypedSPI> T newTypedServiceInstance(final Class<T> typedService, final String type) {
-        Class<?> instanceClass = TYPED_SERVICE_CLASSES.containsKey(typedService) ? TYPED_SERVICE_CLASSES.get(typedService).get(type) : null;
+    public static <T extends TypedSPI> Optional<T> newTypedServiceInstance(final Class<T> typedServiceInterface, final String type) {
+        Class<?> instanceClass = TYPED_SERVICE_CLASSES.containsKey(typedServiceInterface) ? TYPED_SERVICE_CLASSES.get(typedServiceInterface).get(type) : null;
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/spi/ElasticJobServiceLoader.java]: Consider making good use of `Optional#map` if you are going to use `Optional`.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/handler/threadpool/JobExecutorServiceHandlerFactory.java (hunk 1 above)
Comment (wwj-go) [reply elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/handler/threadpool/JobExecutorServiceHandlerFactory.java]: This is a redundant check and will be replaced with a simple Optional#get , as will JobShardingStrategyFactory#getStrategy

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/spi/ElasticJobServiceLoader.java (hunk 2 above)
Comment (wwj-go) [reply elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/spi/ElasticJobServiceLoader.java]: That looks good.

**Pull request closed without merging**